app/src/main/java/com/github/jorgecastillo/kotlinandroid/domain/model/CharacterError.kt

```diff
@@ -13,7 +15,7 @@ package com.github.jorgecastillo.architecturecomponentssample.model.error
  * bring not required complexity to the architecture introducing asynchronous semantics.
  */
 sealed class CharacterError {
-  class AuthenticationError : CharacterError()
-  class NotFoundError : CharacterError()
-  class UnknownServerError : CharacterError()
+  object AuthenticationError : CharacterError()
```

Comment:
What's the benefit on using objects on sealed class elements? Do we need them to be singleton for any reason? They are not storing any shared state that we need to reuse every time, do they?